### ISSUE TITLE
[BUG] all-negative Stacked horizontal Bar Chart size is not correct

### ISSUE THREAD

**Issue by liuxuan30** [danielgindi/ios-charts#222]
Hi Daniel,
I have updated the code to today's latest, after you fixed some of the bugs, thanks for it.

If I use all negative values for stacked horizontal bar chart in demo code like:
```Objective-C
    [yValues addObject:[[BarChartDataEntry alloc] initWithValues:@[ @-10, @-10 ] xIndex: 0]];
    [yValues addObject:[[BarChartDataEntry alloc] initWithValues:@[ @-12, @-13 ] xIndex: 1]];
    [yValues addObject:[[BarChartDataEntry alloc] initWithValues:@[ @-15, @-15 ] xIndex: 2]];
    [yValues addObject:[[BarChartDataEntry alloc] initWithValues:@[ @-17, @-17 ] xIndex: 3]];
    [yValues addObject:[[BarChartDataEntry alloc] initWithValues:@[ @-19, @-20 ] xIndex: 4]];
    [yValues addObject:[[BarChartDataEntry alloc] initWithValues:@[ @-19, @-19 ] xIndex: 5]];
    [yValues addObject:[[BarChartDataEntry alloc] initWithValues:@[ @-16, @-16 ] xIndex: 6]];
    [yValues addObject:[[BarChartDataEntry alloc] initWithValues:@[ @-13, @-14 ] xIndex: 7]];
    [yValues addObject:[[BarChartDataEntry alloc] initWithValues:@[ @-10, @-11 ] xIndex: 8]];
    [yValues addObject:[[BarChartDataEntry alloc] initWithValues:@[ @-5, @-6 ] xIndex: 9]];
    [yValues addObject:[[BarChartDataEntry alloc] initWithValues:@[ @-1, @-2 ] xIndex: 10]];
```
The size of the stack is out of bound, looks like the range is not calculated correctly; the negative range is still 21M, while it should be @-19 + @-20? check the screenshot:
![stacknega](https://cloud.githubusercontent.com/assets/4375169/8719204/669791fe-2bdc-11e5-9b9b-03c6fab93e60.png)

**Comment by danielgindi** [comment]
Did you specify a custom range...?

**Comment by danielgindi** [comment]
The *negative* demo sets a custom axis range.

**Comment by liuxuan30** [comment]
I don't realize you have set
```Objective-C
 _chartView.rightAxis.customAxisMax = 25.0;
 _chartView.rightAxis.customAxisMin = -25.0;
```
Comment it out, it works fine with the all negative values.

However, If I use the default demo code data, e.g. half negative half positive and comment out customAxisMin/Max code, the chart becomes weird... Can you try it on your side?

**Comment by liuxuan30** [comment]
I don't realize you have set
```Objective-C
 _chartView.rightAxis.customAxisMax = 25.0;
 _chartView.rightAxis.customAxisMin = -25.0;
```
Comment it out, it works fine with the all negative values.

However, If I use the default demo code data, e.g. half negative half positive and comment out customAxisMin/Max code, the chart becomes weird... Can you try it on your side?

**Comment by liuxuan30** [comment]
This is the dataSet:
```Objective-C
    [yValues addObject:[[BarChartDataEntry alloc] initWithValues:@[ @-10, @10 ] xIndex: 0]];
    [yValues addObject:[[BarChartDataEntry alloc] initWithValues:@[ @-12, @13 ] xIndex: 1]];
    [yValues addObject:[[BarChartDataEntry alloc] initWithValues:@[ @-15, @15 ] xIndex: 2]];
    [yValues addObject:[[BarChartDataEntry alloc] initWithValues:@[ @-17, @17 ] xIndex: 3]];
    [yValues addObject:[[BarChartDataEntry alloc] initWithValues:@[ @-19, @20 ] xIndex: 4]];
    [yValues addObject:[[BarChartDataEntry alloc] initWithValues:@[ @-19, @19 ] xIndex: 5]];
    [yValues addObject:[[BarChartDataEntry alloc] initWithValues:@[ @-16, @16 ] xIndex: 6]];
    [yValues addObject:[[BarChartDataEntry alloc] initWithValues:@[ @-13, @14 ] xIndex: 7]];
    [yValues addObject:[[BarChartDataEntry alloc] initWithValues:@[ @-10, @11 ] xIndex: 8]];
    [yValues addObject:[[BarChartDataEntry alloc] initWithValues:@[ @-5, @6 ] xIndex: 9]];
    [yValues addObject:[[BarChartDataEntry alloc] initWithValues:@[ @-1, @2 ] xIndex: 10]];
```
and make sure comment out 
```Objective-C
 _chartView.rightAxis.customAxisMax = 25.0;
 _chartView.rightAxis.customAxisMin = -25.0;
```
turn off startAtZeroEnabled:
```Objective-C
_chartView.rightAxis.startAtZeroEnabled = NO;
```
Screenshot:
![half](https://cloud.githubusercontent.com/assets/4375169/8738671/5128686e-2c65-11e5-9c5a-2a265d977dc7.png)

**Comment by liuxuan30** [comment]
@danielgindi, looks like this issue has been fixed on 2.1.2?
